### PR TITLE
Added check when parsing planning tab to verify viability of step based upon progress cell.

### PR DESCRIPTION
**Pull request by bryanjnelson**

### CODE REVIEW

File: src/components/product/product-extractor.service.ts [hunk 1]
```diff
@@ -44,17 +50,26 @@ export class ProductExtractor {
...
+    const rowIndex = row.a1 - sheet.goals.start.row.a1;
+    const progressRow = pnp.progress.goals.start.row.a1 + rowIndex;
+
     const steps = entries(stepColumns).flatMap(([step, column]) => {
       const fiscalYear = sheet.cell(column, row).asNumber;
       const fullFY = fiscalYear ? fullFiscalYear(fiscalYear) : undefined;
       // only include step if it references a fiscal year within the project
       if (!fullFY || !sheet.projectFiscalYears.intersection(fullFY)) {
         return [];
       }
```
Comment (CarsonF) [inline src/components/product/product-extractor.service.ts:66]: This logic is duplicated now in progress extraction. Would you mind abstracting to `isGoalStepPlannedInsideProject` function?

Also here we use `sheet.projectFiscalYears` but in progress we use `sheet.projectDateRange`. I believe this one is correct and we should standardize on that.

**Review by CarsonF — APPROVED** [review]
Otherwise looks good!

**Comment by bryanjnelson** [comment]
@CarsonF - Once this gets your okay I'll squash these together.

**Review by bryanjnelson — COMMENTED** [review]
Love it; something didn't sit well with me about how I had it either.

**Review by CarsonF — APPROVED** [review]

**Pull request merged**